test/Tester/StreamingTests/SampleStreamingTests.cs

```diff
@@ -104,6 +106,44 @@ public SampleStreamingTests(string streamProvider, Logger logger, TestCluster cl
             this.cluster = cluster;
         }
 
+        public async Task StreamingTests_Consumer_Producer_AttachObserver(Guid streamId)
+        {
+            // consumers joins first, producer later
+            var tasks = new List<Task>();
+            var consumers = new List<IStateless_ConsumerGrain>();
+            var subResultGrain = this.cluster.GrainFactory.GetGrain<Tester.StreamingTests.ISubscribeResultGrain>(
+                Tester.StreamingTests.ProgrammaticSubcribingBoostrapper.SubscribeResultGrainId);
+            var subscriptions = await subResultGrain.GetSubscriptionResult();
+            foreach (var sub in subscriptions)
+            {
+                var consumer = this.cluster.GrainFactory.GetGrain<IStateless_ConsumerGrain>(sub.GrainId.PrimaryKey);
+                consumers.Add(consumer);
+                tasks.Add(consumer.AttachObserver(streamId, StreamNamespace, sub.StreamProviderName, GuidId.GetGuidId(sub.SubscriptionId)));
+            }
```

Comment:
AttachObserver should be in grain's onactivate logic to take advantage of programmatic streaming benefits. I will think of something to revise this test